DIPs/DIP1xxx-JS.md

```diff
@@ -0,0 +1,109 @@
+features. Each deprecation ends being handled slightly differently depending on
+who's handling the pull requests. Standardizing the process makes sure that
+deprecations are done very publicly and carefully, as to minimize breakage and
+to provide clear fixes for user code.
+
+## Description
+
+The following lays out the rules to follow to deprecate a part of D.
+
+A symbol or feature must not be marked for removal on a specific date, but rather on a
+specific release. This allows users to easily know if upgrading will break their
+code or not.
+
+Users must be given at least eight non-patch releases before the deprecated features
+are removed. More releases can be given if the removed code is commonly used.
+There are two cases where the deprecation period is allowed to be shorter:
+
+1. The code or feature is notably dangerous or unsafe, and users need to remove
+it from their code as soon as possible.
+2. The existance of the current code precludes its own fix or the fix of an equally
+important issue.
+
+Shorting the deprecation preiod should be done with caution to avoid giving D
+an image of instability.
+
+At the time of the pull request for deprecation, all code in Phobos, Druntime,
+and DMD must be updated to remove use of the effected code. Any projects that
+are tested on the Project Tester that are broken should also have their
+maintainers notified.
+
+Both at the time of deprecation and removal, a changelog entry must be made. This
+changelog entry should have a short motivation for the deprecation (or removal)
+and should to describe what steps can be taken by the user to upgrade their codebase.
+
+In order to facilitate on schedule deprecations, a comment of the format
+`@@@DEPRECATED_[version]@@@` should be added to the top of the code to be removed/disabled.
+This comment allows code to be easily searched before every release to
+catch all planned deprecations.
+
+### Public Functions, Types, and Modules
+
+All removals (or changes that make the code `private`/`package`) of public functions,
+types, and modules must be accompanied with a deprecation period.
+
+The symbol(s) must be marked using the `deprecated` keyword with a message containing
+the planned removal period and/or a pointer to more information pertaining to the
+deprecation. The documentation of the symbol(s) must be updated noting the
+deprecation and removal plan. The documentation should contain information to help
+the users using the symbol(s) transition their code away from the symbol(s).
+
+If the deprecation is occuring because the symbol(s) are being replaced by new
+symbols, both the old and the new symbol(s) should be availible un-deprecated
+in at least one release to allow users to build their code without issue on
+both the `stable` and `master` branches.
+
+On the fourth release in the deprecation period, the documentation for the symbol
+should be removed while keeping the code itself public until removal.
```

Comment:
How about adding a global summary table of the required actions?
Quick proposal:

action | when
---------| --------
add deprecation warning | immediately
add deprecation message to the docs | immediately
add deprecation changelog | immediately
remove deprecation from the docs | Fourth release
remove symbol + add final removal warnings | Fifth release